scripts/esbuild/util.ts

```diff
@@ -0,0 +1,88 @@
+import { join } from 'path';
+import { BuildOptions } from '../utils/options';
+
+/**
+ * Aliases which map the module identifiers we set in `paths` in `tsconfig.json` to
+ * bundles (built either with esbuild or with rollup).
+ *
+ * @param opts options for the current build
+ * @returns a map of aliases to bundle entry points, relative to the root directory
+ */
+export function getEsbuildAliases(opts: BuildOptions): Record<string, string> {
+  return {
+    // node module redirection
+    chalk: 'ansi-colors',
+
+    // mapping aliases to top-level bundle entrypoints
+    '@stencil/core/testing': './testing/index.js',
+    '@stencil/core/compiler': './compiler/stencil.js',
+    '@stencil/core/dev-server': './dev-server/index.js',
+    '@stencil/core/mock-doc': './mock-doc/index.cjs',
+    '@stencil/core/internal/testing': './internal/testing/index.js',
```

Comment:
set up aliases which will point to the entry points of other bundles. Note that in combination with `getEsbuildExternalModules`, below, these will be marked as external and thus imported or required instead of being bundled in